### PR TITLE
Misc sci@ bumps

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

**Comment by mm1ke** [comment]
Out of curiosity. I hope that's fine to "start a review" - or is it better to leave just comments instead? I didn't made reviews yet so i'm not sure what is preferred?

**Comment by jsmolic** [comment]
@mm1ke However you like, if you have multiple comments you can add them in a review, but up to you. Any comment/review/suggestion is always welcome :)

File: sci-libs/alglib/alglib-3.17.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+SLOT="0/3.8"
+LICENSE="GPL-2+"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+src_prepare() {
+	cp "${FILESDIR}"/CMakeLists.txt-3.8.2 CMakeLists.txt || die
+	use x86 && append-cppflags -ffloat-store
```
Comment (SoapGentoo) [inline sci-libs/alglib/alglib-3.17.0.ebuild:21]: do this in `src_configure`

File: sci-libs/cminpack/cminpack-1.3.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+LICENSE="minpack"
+SLOT="0/1"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc test"
+RESTRICT="!test? ( test )"
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMINPACK_LIB_INSTALL_DIR=$(get_libdir)
+		-DBUILD_SHARED_LIBS=ON
```
Comment (SoapGentoo) [inline sci-libs/cminpack/cminpack-1.3.8.ebuild:21]: not needed in EAPI 7

File: sci-libs/cmpfit/cmpfit-1.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+SRC_URI="http://www.physics.wisc.edu/~craigm/idl/down/${P}.tar.gz"
+
+LICENSE="public-domain minpack"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+DOCS=( README DISCLAIMER )
+
+src_compile() {
+	$(tc-getCC) ${CFLAGS} -fPIC -c mpfit.c -o mpfit.so || die "PIC compilation failed"
```
Comment (SoapGentoo) [inline sci-libs/cmpfit/cmpfit-1.4.ebuild:19]: `CPPFLAGS`?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-libs/alglib/alglib-3.17.0.ebuild (hunk 1 above)
Comment (jsmolic) [reply sci-libs/alglib/alglib-3.17.0.ebuild]: done

File: sci-libs/cminpack/cminpack-1.3.8.ebuild (hunk 2 above)
Comment (jsmolic) [reply sci-libs/cminpack/cminpack-1.3.8.ebuild]: done

File: sci-libs/cmpfit/cmpfit-1.4.ebuild (hunk 3 above)
Comment (jsmolic) [reply sci-libs/cmpfit/cmpfit-1.4.ebuild]: done

File: sci-libs/cminpack/cminpack-1.3.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+SRC_URI="https://github.com/devernay/cminpack/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="minpack"
+SLOT="0/1"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc test"
+RESTRICT="!test? ( test )"
+
+src_configure() {
+	local mycmakeargs=(
```
Comment (a17r) [inline sci-libs/cminpack/cminpack-1.3.8.ebuild:19]: Please check out my bump of this version in commit 71c501b2822cffd4da67aa754925f7ce6d002288, you were missing a new default enabled option here.

I also had to patch CMakeLists.txt to make it install its cmake config into the usual path.

Either drop the bump or rebase and work on adding a new `IUSE` here (I just disabled it for now).

File: sci-libs/alglib/alglib-3.17.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+inherit cmake flag-o-matic
+
+DESCRIPTION="Numerical analysis and data processing library"
+HOMEPAGE="https://www.alglib.net/"
+SRC_URI="https://www.alglib.net/translator/re/${P}.cpp.gpl.tgz"
+S="${WORKDIR}"/cpp
+
+SLOT="0/3.8"
+LICENSE="GPL-2+"
```
Comment (a17r) [inline sci-libs/alglib/alglib-3.17.0.ebuild:14]: Those lines should switch places.

**Review by a17r — CHANGES_REQUESTED** [review]

File: sci-libs/alglib/alglib-3.17.0.ebuild (hunk 5 above)
Comment (jsmolic) [reply sci-libs/alglib/alglib-3.17.0.ebuild:14]: yup, nice catch

File: sci-libs/cminpack/cminpack-1.3.8.ebuild (hunk 4 above)
Comment (jsmolic) [reply sci-libs/cminpack/cminpack-1.3.8.ebuild:19]: Thanks, I missed it somehow. I'll just drop this commit we can revisit blas later

File: sci-calculators/qalculate-gtk/qalculate-gtk-3.17.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+BDEPEND="
+	app-text/rarian
+	dev-util/intltool
+	sys-devel/gettext
+	virtual/pkgconfig"
+
+src_prepare() {
+	# Required by src_test() and `make check`
+	echo data/calendarconversion.ui > po/POTFILES.skip || die
+	echo data/periodictable.ui >> po/POTFILES.skip || die
```
Comment (SoapGentoo) [inline sci-calculators/qalculate-gtk/qalculate-gtk-3.17.0.ebuild:33]: use the `cat` pattern here

File: sci-libs/alglib/alglib-3.17.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+src_prepare() {
+	cp "${FILESDIR}"/CMakeLists.txt-3.8.2 CMakeLists.txt || die
+	cmake_src_prepare
+}
+
+src_configure() {
+	use x86 && append-cppflags -ffloat-store
```
Comment (SoapGentoo) [inline sci-libs/alglib/alglib-3.17.0.ebuild:25]: is this really a `CPPFLAGS` options? :thinking:

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-calculators/qalculate-gtk/qalculate-gtk-3.17.0.ebuild (hunk 6 above)
Comment (jsmolic) [reply sci-calculators/qalculate-gtk/qalculate-gtk-3.17.0.ebuild]: done

File: sci-libs/alglib/alglib-3.17.0.ebuild (hunk 7 above)
Comment (jsmolic) [reply sci-libs/alglib/alglib-3.17.0.ebuild]: I think it's a cflag

**Review by SoapGentoo — APPROVED** [review]

**Pull request closed without merging**